pages/api/address/transactions/[address].ts

```diff
@@ -32,22 +33,34 @@ export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>
       if (req.query.address === '' || req.query.address === undefined) {
         throw new Error(ADDRESS_NOT_PROVIDED_400.message)
       }
+
       const address = parseAddress(req.query.address as string)
+
+      // this flag ?serverOnly=1 tells us to only retrieve what we have in the database
+      const serverOnly = req.query.serverOnly === '1'
+
+      if (!await addressExistsBySubstring(address)) {
+        if (serverOnly) throw new Error(NO_ADDRESS_FOUND_404.message)
+
+        await upsertAddress(address)
+        await syncTransactionsAndPricesForAddress(address)
+      }
       const transactions = await fetchAddressTransactions(address)
+
       res.status(200).send(transactions)
     } catch (err: any) {
       switch (err.message) {
         case ADDRESS_NOT_PROVIDED_400.message:
           res.status(ADDRESS_NOT_PROVIDED_400.statusCode).json(ADDRESS_NOT_PROVIDED_400)
           break
-        case NO_ADDRESS_FOUND_404.message: {
-          res.status(NO_ADDRESS_FOUND_404.statusCode).json(NO_ADDRESS_FOUND_404)
-          break
-        }
         case INVALID_ADDRESS_400.message: {
           res.status(INVALID_ADDRESS_400.statusCode).json(INVALID_ADDRESS_400)
           break
         }
+        case NO_ADDRESS_FOUND_404.message: {
+          res.status(NO_ADDRESS_FOUND_404.statusCode).send(NO_ADDRESS_FOUND_404)
```

Comment:
Better stick to `.json` which is being used in the other cases

Reply:
done